pkg/platform/api/platform.go

```diff
@@ -0,0 +1,146 @@
+package api
+
+import (
+	"github.com/Masterminds/semver"
+	"github.com/bottlerocket-os/bottlerocket-update-operator/pkg/logging"
+	"github.com/bottlerocket-os/bottlerocket-update-operator/pkg/platform"
+	"github.com/pkg/errors"
+)
+
+// Assert Update-API as a platform implementor.
+var _ platform.Platform = (*Platform)(nil)
+
+type Platform struct {
+	log        logging.Logger
+	apiClient  *APIClient
+}
+
+func New() (*Platform, error) {
+	return &Platform{log: logging.New("platform"), apiClient: NewAPIClient()}, nil
+}
+
+type statusResponse struct {
+	osVersion *semver.Version
+}
+
+func (sr *statusResponse) OK() bool {
+	// Bottlerocket OS version needs to be at least a certain version to support the Update API
+	constraint, _ := semver.NewConstraint(">= " + minimumRequiredOSVer)
+	return constraint.Check(sr.osVersion)
+}
+
+func (p Platform) Status() (platform.Status, error) {
+	// Try to determine if the update API is supported in the Bottlerocket host
+	osInfo, err := p.apiClient.GetOSInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	osVersion, err := semver.NewVersion(osInfo.VersionID)
+	p.log.Info("current running OS version: ", osInfo.VersionID)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse 'version_id' field as semver")
+	}
+	return &statusResponse{osVersion: osVersion}, nil
+}
+
+type listAvailableResponse struct {
+	chosenUpdate *UpdateImage
+}
+
+func (lar *listAvailableResponse) Updates() []platform.Update {
+	if lar.chosenUpdate == nil {
+		return nil
+	}
+	updates := make([]platform.Update, 1)
+	updates[0] = lar.chosenUpdate
+	return updates
+}
+
+func (p Platform) ListAvailable() (platform.Available, error) {
+	p.log.Debug("fetching list of available updates")
+
+	// Refresh list of updates and check if there are any available
+	err := p.apiClient.RefreshUpdates()
+	if err != nil {
+		return nil, err
+	}
+
+	updateStatus, err := p.apiClient.GetUpdateStatus()
+	if err != nil {
+		return &listAvailableResponse{}, err
```

Comment:
Is there a particular reason why you can't return `nil, err` in all of the error cases?